unikraft/app/loader.go

```diff
@@ -100,6 +100,12 @@ func NewApplicationFromInterface(ctx context.Context, iface map[string]interface
 		return nil, err
 	}
 
+	for _, target := range app.targets {
+		if target.Architecture().Name() == "" || target.Platform().Name() == "" {
+			return nil, fmt.Errorf("architecture and platform must be specified for all targets")
+		}
+	}
+
```

Comment:
We could do this earlier at the end of [`target.TransformFromSchema`](https://github.com/unikraft/kraftkit/blob/staging/unikraft/target/transform.go#L118) method, saving us having to run this loop. Then you can also two if error checks, one for plat and the other for arch.

Reply:
Fixed and also added a check for the case where you would do `- string` with no `/`, which would result in a sigsegv